test/src/test/java/hudson/util/ProcessTreeKillerTest.java

```diff
@@ -35,35 +39,57 @@
     @Rule
     public JenkinsRule j = new JenkinsRule();
     private Process process;
-    
+
     @After
     public void tearDown() throws Exception {
         ProcessTree.vetoersExist = null;
         if (null != process)
             process.destroy();
-    }    
-    
-    @Test
-	public void manualAbortProcess() throws Exception {
-		ProcessTree.enabled = true;
-		FreeStyleProject project = j.createFreeStyleProject();
+    }
 
-		// this contains a maven project with a single test that sleeps 5s.
-		project.setScm(new ExtractResourceSCM(getClass().getResource(
-				"ProcessTreeKiller-test-project.jar")));
-		project.getBuildersList().add(new Maven("install", "maven"));
+    @Test
+    public void manualAbortProcess() throws Exception {
+        ProcessTree.enabled = true;
+        FreeStyleProject project = j.createFreeStyleProject();
+
+        // this contains a maven project with a single test that sleeps 5s.
+        project.setScm(new ExtractResourceSCM(getClass().getResource(
+                "ProcessTreeKiller-test-project.jar")));
+        project.getBuildersList().add(new Maven("install", "maven"));
+
+        // build the project, wait until tests are running, then cancel.
+        final QueueTaskFuture<FreeStyleBuild> f = project.scheduleBuild2(0);
+        final FreeStyleBuild b = f.waitForStart();
+
+        // wait until maven starts sleep test
+        boolean sleepTestStarted = false;
+        while (b.isBuilding()) {
+            final String log = readFully(b.getLogReader());
+            sleepTestStarted = log != null && log.contains("Running SleepingTest");
+            if (sleepTestStarted) {
+                break;
+            }
+
+            Thread.sleep(1000);
```

Comment:
Nit: usually easier to read by using `TimeUnit.SECONDS.sleep(1);` instead.

Reply:
Okay, fixed